packages/realm-web/src/auth-providers/index.ts

```diff
@@ -0,0 +1,37 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Transport } from "../transports";
+
+import { EmailPasswordAuthProvider } from "./EmailPasswordAuthProvider";
+import { ApiKeyAuthProvider } from "./ApiKeyAuthProvider";
+
+// TODO: Consider if we should query the service for enabled authentication providers before creating clients.
```

Comment:
why should client creation depend on server settings. if a client is created and used then it should fail if it can't reach its endpoint. or maybe I am not fully understanding the comment.

Reply:
No you're right, they will just fail (with 404s). It's just that it might be a better developer experience if these objects were not there when the server-side of the API was missing completely (due to the setup of the service). Also just a sidenote that I wanted to revisit later.

Reply:
I think doing that will be strange, that's why I commented on the TODO. Isn't this like a web browser stopping you from using  `https://` if you don't have https enabled on the server :) also this might be a point for security breach or a denial of service